docs/Rest-API.md

```diff
@@ -0,0 +1,147 @@
+# Axe REST API
+
+Documentation of the REST API that comes along with the Axe service.
+
+## AxeResult
+
+This endpoint serves as a wrapper to collect all of the information from an axe run. 
+
+### Motivation
+
+- Provide access to results and context easily.
+- Avoid timing issues that come with multiple requests.
+
+### Simplified Sample
+```
+{
+  "axeContext" : { @AxeContext },
+  "axeResult" : { AXRL },
+  "axeConf" : { TBD by Framework }  
+}
+```
+
+### Requirements
+
+- Data points are collected at the same point in time, or as close as can be expected.
+
+## AxeContext
+
+Wraps data together that is needed to identify a particular View, with a particular screenshot,
+for a particular device configuration.
+
+### Motivation
+
+- Collect View information in one place.
+- Allow for direct correlation of screenshot and ViewHierarchy
+- Easy developer debugging.
+- Reproducability of results.
+
+### Simplified Sample
+
+```
+{
+  "axeView" : @AxeView,
+  "axeDevice" : @AxeDevice,
+  "axeEvents" : @AxeEvents,
+  "screenshot" : "base64png data"
+}
+```
+
+## AxeView
+
+Serves up properties about the Accessibility Hierarchy of a particular view. 
+
+### Motivation
+
+- All motivations of AxeContext apply.
+- Rule debugging.
+- Simplified test case collection.
+- Simplified false positive identification.
+
+### Simplified Sample
+
+```
+{
+  "Arbitrary Property Name": "value1",
+  "Arbitrary Property Name2": "value2",
+  "children" : [
+    {
+      "Arbitrary Property Name": "value3",
+      "Arbitrary Property Name2": "value24"
+      "children": []
+    }, 
+    {
+      "Arbitrary Property Name": "value3",
+      "Arbitrary Property Name2": "value24",
+      "children" : []
+    }
+  ]
+}
+```
+
+### Requirements
+
+- The Arbitrary property names of the children are identical to the root. 
+- If a property is omitted from a child it is assumed to be not present or NULL.
+- Arbitrary property names are NOT calculated, but are values present in the respective OS APIs.
+
+## AxeDevice
+
+Collect prescriptive information that helps identify what type of context various information 
+was collected from.
+
+### Motivation
+
+- Identify device version and operating system.
+- Provide metrics for screen size so view hierarchy can be correlated.
+
+### Simplified Sample
+
+```
+{
+  "os": {
```

Comment:
>os [](start = 3, length = 2)

You might want to include a property bag for platform-specific stuff that may neatly fit into the fields that you're defining here.